test/recipe_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+)
+
+var testFilePaths = []string{
+	"ingress/single-cluster/",
+}
+
+func TestRecipe(t *testing.T) {
+	for _, fp := range testFilePaths {
+		dirs, err := os.ReadDir(fp)
+		if err != nil {
+			t.Fatalf("Failed to open directory %s: %v", fp, err)
+		}
+
+		for _, dir := range dirs {
+			dir := dir
+			t.Run(dir.Name(), func(t *testing.T) {
+				t.Parallel()
+				setupFile := path.Join(fp, dir.Name(), "setup.sh")
```

Comment:
You can use a for loop to avoid redundant code.

**do not** assume err means not found/missing. You should print the actual error that is being return from stat()

```
for _, file := range []string{"setup.sh", "run-test.sh", "cleanup.sh"} {
  path := path.Join(fp, dir.Name(), file)
  if _, err := os.Stat(file); err != nil {
    t.Skipf("Skipping test %q: stat(%q) = %v", dir.Name(), path, err)
  }
}

```